lib/DBD/mysql.pm

```diff
@@ -1186,6 +1186,18 @@ flags:
 These are used to specify the respective parameters of a call
 to mysql_ssl_set, if mysql_ssl is turned on.
 
+=item mysql_ssl_verify_server_cert
+
+This turns on the MYSQL_OPT_SSL_VERIFY_SERVER_CERT connection option.
+
+Note that this is deprecated as of MySQL 5.7.11 and is removed in MySQL 8.0
+
+=item mysql_ssl_mode
+
+Can be set to 'disabled', 'preferred', 'required', 'verify_ca' or 'verify_identity'.
+Default is 'preferred'.
+
+Available from MySQL 5.7.11
```

Comment:
This config logic is bad :-(

When you set `mysql_ssl=1` then SSL is `required`. And when you set `mysql_ssl_verify_server_cert=1` then mode should be `verify_identity`.

Rather do not export low-level libmysqlclient flags and configs to high-level DBD::mysql directly, because there is place for specifying nonsense combination like "mysql_ssl=1;mysql_ssl_mode=disabled"... Also I have feeling that with different MySQL and MariaDB versions I would be able to set probably same SSL/TLS options but with different low-level libmysqlclient functions and flags. And I think that DBD::mysql should not export directly those low-level flags but rather wrap them into stable API and did all C client logic in its own code... Think about application developer who want to use DBD::mysql. He really do not want to add into his code different if/else cases for different MySQL and MariaDB versions to correctly set SSL options.

Option `mysql_ssl_verify_server_cert` is already supported by DBD::mysql, but documentation is missing. This should be fixed first and probably support this option also for MySQL 8.0 (emulated by DBD::mysql).

What about introducing option `mysql_ssl_prefer` (with default value true) which will be equivalent for `SSL_MODE_PREFERRED` instead whole option `mysql_ssl_mode`? Becaue `SSL_MODE_DISABLED` is equivalent for `mysql_ssl=0`, `SSL_MODE_REQUIRED` for `mysql_ssl=1`, `SSL_MODE_VERIFY_IDENTITY` for `mysql_ssl_verify_server_cert=1` and `SSL_MODE_VERIFY_CA` is probably not needed...